rs/rosetta-api/icrc1/ledger/src/main.rs

```diff
@@ -135,18 +142,79 @@ fn pre_upgrade() {
         .expect("failed to write instructions consumed to stable memory");
 }
 
+#[cfg(feature = "upgrade-to-memory-manager")]
+#[pre_upgrade]
+fn pre_upgrade() {
+    #[cfg(feature = "canbench-rs")]
+    let _p = canbench_rs::bench_scope("pre_upgrade");
+
+    let start = ic_cdk::api::instruction_counter();
+    UPGRADES_MEMORY.with_borrow_mut(|bs| {
+        Access::with_ledger(|ledger| {
+            let writer = Writer::new(bs, 0);
+            let mut buffered_writer = BufferedWriter::new(8388608, writer);
+            ciborium::ser::into_writer(ledger, &mut buffered_writer)
+                .expect("Failed to write the Ledger state in stable memory");
+            let end = ic_cdk::api::instruction_counter();
+            let instructions_consumed = end - start;
+            let counter_bytes: [u8; 8] = instructions_consumed.to_le_bytes();
```

Comment:
Could we run out of instructions if we transfer the entire memory at once?

Reply:
From @mbjorkqvist calculations, it seems with the BufferedWriter we should consume similar number of instructions as in a regular pre_upgrade, so if this fails, the other pre_upgrade will as well.